CONTRIBUTING.md

```diff
@@ -42,6 +42,8 @@ If additional new Python module dependencies are required, think about where to
 - If they're optional dependencies for additional functionality, then put them in the pyproject.toml file like were done for [flash-attn](https://github.com/foundation-model-stack/fms-hf-tuning/blob/main/pyproject.toml#L44) or [aim](https://github.com/foundation-model-stack/fms-hf-tuning/blob/main/pyproject.toml#L45).
 - If it's an additional dependency for development, then add it to the [dev](https://github.com/foundation-model-stack/fms-hf-tuning/blob/main/pyproject.toml#L43) dependencies.
 
+if using `poetry` to install dependencies, you can optionally leverage [poetry add](https://python-poetry.org/docs/cli/#add) to add new dependencies to pyproject.  
```

Comment:
I feel this is abit confusing. If we have a peoetry flow, and there is a new dep, shouldnt we have some guidance on when it has to be added to the lockfile?

Also some guidance to say that for optional deps it should not be added.

Reply:
true, maybe we can continue adding to pyproject.yml , like we do now and then we have to run `poetry update` ? 
I can change it to that instead

Reply:
There're 2 ways to add a new dependency. One is to use `poetry` i.e. `poetry add` the other is to manually edit the `pyproject.toml` file and then run `poetry update` which will then ensure `poetry.lock` is consistent with pyproject.toml (that's basically what `poetry add` does on your behalf).

If you opt for the 2nd route but forget to run `poetry update` then the next time you run `poetry install` (or the CI/CD runs it) you'll get an exception that the `poetry.lock` and `pyproject.toml` files are inconsistent and that you should run  the `poetry update` command.